### PR TITLE
[Bignum] Add unit tests for ecp_mod_p448

### LINKED ISSUE #7248: Add unit tests for ecp_mod_p448()
The task is to add unit tests for `ecp_mod_p448()`. This function reduces the input modulo `curve448_p`. The result of `ecp_mod_p448()` is not always canonical, use `mpi_mod_mpi()` to get a canonical representative for easier testing.

Only cases where the input is non-negative and at most 896 bit long need to be tested. (This is because soon these will be replaced by their fixed width `_raw` counterparts.)

### PR DESCRIPTION
**Pull request by paul-elliott-arm**
## Description

Add unit tests for curve p448

Resolves #7248

## Gatekeeper checklist

- [x] **changelog** ~~provided, or~~ not required (part of larger feature)
- [x] **backport** ~~done, or~~ not required (new feature)
- [x] **tests** provided, ~~or not required~~

### CODE REVIEW

**Review by minosgalanakis — APPROVED** [review]
Looks good

File: library/ecp_invasive.h [hunk 1]
```diff
@@ -193,20 +193,28 @@ int mbedtls_ecp_mod_p256k1(mbedtls_mpi *N);
...
+MBEDTLS_STATIC_TESTABLE
+int mbedtls_ecp_mod_p448(mbedtls_mpi *N);
+
+#endif /* MBEDTLS_ECP_DP_CURVE448_ENABLED */
+
+/**
+ *  Initialise a modulus with hard-coded const curve data.
  *
- * \note            The caller is responsible for the \p N modulus' memory.
+ * \note The caller is responsible for the \p N modulus' memory.
```
[the remark this answers is not in the document]
Comment (paul-elliott-arm) [reply library/ecp_invasive.h]: I think my editors auto-formatting got a little carried away prior to the last rebase. I have removed this.

**Review by minosgalanakis — DISMISSED** [review]
LGTM

File: scripts/mbedtls_dev/ecp.py [hunk 2]
```diff
@@ -682,3 +682,93 @@ def result(self) -> List[str]:
...
+
+        # 2^448 - 1
+        ("ffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
+         "ffffffffffffffffffffffffffffffffffffffffffffffffffffffff"),
+
+        # Maximum canonical P448 multiplication result
+        ("fffffffffffffffffffffffffffffffffffffffffffffffffffffffd"
+         "ffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
+         "00000000000000000000000000000000000000000000000000000002"
+         "00000000000000000000000000000000000000000000000000000001"),
```
Comment (gabor-mezei-arm) [inline scripts/mbedtls_dev/ecp.py:719]: The maximum canonical multiplication result is  $(P - 1)^2$  (where in this case $P = 2^{448} - 2^{224} - 1$)

```suggestion
        # Maximum canonical P448 multiplication result
        ("fffffffffffffffffffffffffffffffffffffffffffffffffffffffd"
         "fffffffffffffffffffffffffffffffffffffffffffffffffffffffd"
         "00000000000000000000000000000000000000000000000000000004"
         "00000000000000000000000000000000000000000000000000000004"),
```

**Review by gabor-mezei-arm — CHANGES_REQUESTED** [review]

File: scripts/mbedtls_dev/ecp.py (hunk 2 above)
Comment (paul-elliott-arm) [reply scripts/mbedtls_dev/ecp.py:719]: Sorry, that was indeed $(P )^2$ not $(P - 1)^2$

**Review by gabor-mezei-arm — APPROVED** [review]
LGTM

**Review by minosgalanakis — APPROVED** [review]
LGTM

**Pull request merged**